README.md

```diff
@@ -39,7 +39,83 @@ make test
+timing side channels through suitable barriers and constant-time patterns.
+
+Absence of secret-dependent branches, memory-access patterns and variable-latency instructions is also tested using `valgrind`
+with various combinations of compilers and compilation options.
+
+## Design
+
+mldsa-native is split into a _frontend_ and two _backends_ for arithmetic and FIPS202 / SHA3. The frontend is
+fixed, written in C, and covers all routines that are not critical to performance. The backends are flexible, take care of
+performance-sensitive routines, and can be implemented in C or native code (assembly/intrinsics); see
+[mldsa/native/api.h](mldsa/native/api.h) for the arithmetic backend and 
+[mldsa/fips202/native/api.h](mldsa/fips202/native/api.h) for the FIPS-202 backend. mldsa-native currently
+offers backends for C, AArch64, and x86_64 - if you'd like contribute new backends, please reach out or just open a PR.
+
+## Usage
+
+Once mldsa-native reaches production readiness, you will be able to import [mldsa](mldsa) into your project's source tree and build using your preferred build system. See [mldsa](mldsa) for more information. The build system provided in this repository is for development purposes only.
+
+### Will I be able to bring my own FIPS-202?
+
+Yes. mldsa-native will include a FIPS-202 implementation, but you will be able to use your own SHA-3/SHAKE implementation instead if needed. This will follow the same pattern as mlkem-native.
+
+### Will I need to use the assembly backends?
+
+No. If you want a C-only build, you will be able to omit the native backend directories from your import and configure the build accordingly in your config.h.
+
+### Do I need to setup CBMC to use mldsa-native?
+
+No. While we recommend considering formal verification for security-critical applications, mldsa-native will build and run without CBMC. Function contracts and loop invariants will be ignored unless `CBMC` is defined during compilation.
+
+### Does mldsa-native support all security levels of ML-DSA?
+
+Yes. mldsa-native supports all three ML-DSA security levels (ML-DSA-44, ML-DSA-65, ML-DSA-87) as defined in FIPS 204. The security level is a compile-time parameter.
+
+### Does mldsa-native use hedged or deterministic signing?
+
+By default, mldsa-native uses the "hedged" signing variant as specified in FIPS 204 Section 3.4, with `MLD_RANDOMIZED_SIGNING` enabled in [mldsa/config.h](mldsa/config.h). The hedged variant uses both fresh randomness at signing time and precomputed randomness from the private key. This helps mitigate side-channel attacks while protecting against potential flaws in the random number generator.
+
+The deterministic variant can be enabled by undefining `MLD_RANDOMIZED_SIGNING`, but FIPS 204 warns that this should not be used on platforms where side-channel attacks are a concern, as the lack of fresh randomness makes fault attacks more difficult to mitigate.
```

Comment:
I disagree with NIST's classification of fault injection attacks being a type side-channel attack. 
Can we change this to "fault injection attacks and side-channel attacks"? 
Fault injection attacks are actually more critical here because differential fault attacks on ML-DSA are straightforward given a faulty and a valid signature using the same randomness: z' - z = (y + c's) - (y + cs) = (c' - c)s => s.

Reply:
[01f2c95](https://github.com/pq-code-package/mldsa-native/pull/385/commits/01f2c9538c8d1f2255e336efb5924ccd6cc02e63)